packages/local_auth/local_auth/windows/local_auth_plugin.cpp

```diff
@@ -0,0 +1,187 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+#include "include/local_auth/local_auth_plugin.h"
+
+// This must be included before many other Windows headers.
+#include <UserConsentVerifierInterop.h>
+#include <flutter/encodable_value.h>
+#include <flutter/method_channel.h>
+#include <flutter/plugin_registrar_windows.h>
+#include <flutter/standard_method_codec.h>
+#include <pplawait.h>
+#include <ppltasks.h>
+#include <windows.h>
+#include <winrt/Windows.Foundation.h>
+#include <winrt/Windows.Security.Credentials.UI.h>
+#include <winstring.h>
+
+#include <map>
+#include <memory>
+#include <sstream>
+
+namespace {
+
+using namespace flutter;
+using namespace winrt;
+
+template <typename T>
+T GetArgument(const std::string arg, const EncodableValue* args, T fallback) {
+  T result{fallback};
+  const auto* arguments = std::get_if<EncodableMap>(args);
+  if (arguments) {
+    auto result_it = arguments->find(EncodableValue(arg));
+    if (result_it != arguments->end()) {
+      result = std::get<T>(result_it->second);
+    }
+  }
+  return result;
+}
+
+HWND GetRootWindow(flutter::FlutterView* view) {
+  return ::GetAncestor(view->GetNativeWindow(), GA_ROOT);
+}
+
+std::wstring s2ws(const std::string& s) {
+  int len;
+  int slength = (int)s.length() + 1;
+  len = MultiByteToWideChar(CP_ACP, 0, s.c_str(), slength, 0, 0);
+  std::wstring r(len, L'\0');
+  MultiByteToWideChar(CP_ACP, 0, s.c_str(), slength, &r[0], len);
+  return r;
+}
+
+class LocalAuthPlugin : public Plugin {
+ public:
+  static void RegisterWithRegistrar(PluginRegistrarWindows* registrar);
+
+  LocalAuthPlugin(flutter::PluginRegistrarWindows* registrar);
+
+  virtual ~LocalAuthPlugin();
+
+ private:
+  flutter::PluginRegistrarWindows* registrar_;
+
+  // Called when a method is called on this plugin's channel from Dart.
+  void HandleMethodCall(const MethodCall<EncodableValue>& method_call,
+                        std::unique_ptr<MethodResult<EncodableValue>> result);
+
+  winrt::fire_and_forget Authenticate(
```

Comment:
Inside this method, I'm only calling a Windows Runtime IAsyncXxx method, so C++/WinRT guarantees that it will capture the calling context and ensure I'm still on it when the continuation resumes, so I'm only calling result-> anything on the same thread where each method was called from.

Reply:
Thanks for the explanation; I haven't really had any experience with WinRT so misunderstood the behavior.